packages/domains-table/src/domains-table-header/columns.ts

```diff
@@ -42,23 +43,23 @@ export const allSitesViewColumns = (
 		isSortable: true,
 		initialSortDirection: 'asc',
 		supportsOrderSwitching: true,
-		sortFunctions: [ getSimpleSortFunctionBy( 'domain' ) ],
+		sortFunctions: [ getSimpleSortFunctionBy( 'owner' ) ],
 	},
 	{
 		name: 'site',
 		label: __( 'Site', __i18n_text_domain__ ),
 		isSortable: true,
 		initialSortDirection: 'asc',
 		supportsOrderSwitching: true,
-		sortFunctions: getSiteSortFunctions(),
+		sortFunctions: [ getSimpleSortFunctionBy( 'blog_name' ) ],
 	},
 	{
 		name: 'expire_renew',
 		label: __( 'Expires / renews on', __i18n_text_domain__ ),
 		isSortable: true,
 		initialSortDirection: 'asc',
 		supportsOrderSwitching: true,
-		sortFunctions: [ getSimpleSortFunctionBy( 'expiry' ), getSimpleSortFunctionBy( 'domain' ) ],
+		sortFunctions: [ getSimpleSortFunctionBy( 'expiry' ) ],
```

Comment:
Falling back to `domain` or `blog_name` doesn't seem unreasonable. Many domain types don't have an expiry. Registered with external providers for instance. There could be lots, having a consistent secondary sort field seems good.

Reply:
If there is no expiry, then can't we just have it handled as null/undefined? It feels random to me to use the domain. In the UI we show `-` but if we have a fallback for `domain` then it looks weird, IMO.

Reply:
>There could be lots, having a consistent secondary sort field seems good.

It would be good to sort nulls based on a secondary sort, but that's not what we were doing, which is why I think it was confusing. We were doing everything in one sort and the second sort only got used if the result was `0`.  Instead, it should have been to sort null and non-null separately and then join the two results.

At this stage, I think showing all nulls at the start or end unsorted is fine for i1.